explorer/interpreter/interpreter.cpp

```diff
@@ -788,22 +816,23 @@ auto Interpreter::CallFunction(const CallExpression& call,
     }
     case Value::Kind::BoundMethodValue: {
       const auto& m = cast<BoundMethodValue>(*fun);
-      const FunctionDeclaration& method = m.declaration();
-      CARBON_CHECK(method.is_method());
+      const FunctionDeclaration* method =
+          dyn_cast<FunctionDeclaration>(&m.declaration());
```

Comment:
Can this be restored to the original code, just `const FunctionDeclaration& method = m.declaration();`?

Reply:
oh yes!